webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/KnoraUserService.scala

```diff
@@ -203,17 +203,24 @@ case class KnoraUserService(
   def addUserToProjectAsAdmin(
     user: KnoraUser,
     project: Project,
-  ): IO[IsProjectAdminMember | NotProjectMember, KnoraUser] = for {
-    _ <- ZIO.fail(IsProjectAdminMember(user.id, project.id)).when(user.isInProjectAdminGroup.contains(project.id))
-    _ <- ZIO.fail(NotProjectMember(user.id, project.id)).unless(user.isInProject.contains(project.id))
-    _ <- addUserToProjectAsAdmin(user, project.id)
-  } yield user
+  ): IO[IsProjectAdminMember | NotProjectMember, KnoraUser] =
+    addUserToProjectAsAdmin(user, project.id)
 
-  def addUserToProjectAsAdmin(user: KnoraUser, project: KnoraProject): UIO[KnoraUser] =
+  def addUserToProjectAsAdmin(
+    user: KnoraUser,
+    project: KnoraProject,
+  ): IO[IsProjectAdminMember | NotProjectMember, KnoraUser] =
     addUserToProjectAsAdmin(user, project.id)
 
-  private def addUserToProjectAsAdmin(user: KnoraUser, projectIri: ProjectIri): UIO[KnoraUser] =
-    updateUser(user, UserChangeRequest(projectsAdmin = Some(user.isInProjectAdminGroup :+ projectIri))).orDie
+  private def addUserToProjectAsAdmin(
+    user: KnoraUser,
+    projectIri: ProjectIri,
+  ): IO[IsProjectAdminMember | NotProjectMember, KnoraUser] = for {
+    _      <- ZIO.fail(IsProjectAdminMember(user.id, projectIri)).when(user.isInProjectAdminGroup.contains(projectIri))
+    _      <- ZIO.fail(NotProjectMember(user.id, projectIri)).unless(user.isInProject.contains(projectIri))
+    request = UserChangeRequest(projectsAdmin = Some((user.isInProjectAdminGroup :+ projectIri).distinct))
```

Comment:
```suggestion
    request = UserChangeRequest(Some((user.isInProjectAdminGroup :+ projectIri).distinct))
```

Reply:
This would introduce a bug and set `projects` instead of the `projectsAdmin` property.